### PR TITLE
Replace hardcoded prompt with Jinja template

### PR DESCRIPTION
**Pull request by joelwiesmann**
Replace hardcoded vector query engine prompt with Jinja template.

### CODE REVIEW

File: griptape/engines/query/vector_query_engine.py [hunk 1]
```diff
@@ -28,28 +33,26 @@ def query(
...
-        ]
-
-        metadata = f"Metadata: {metadata if metadata else 'no metadata available'}"
-        prefix = f"{' '.join(prefix_list)}\n\n{metadata}"
-        question = f"\n\nQuestion: {query}"
-        answer = "\n\nAnswer: "
+        prefix = []
 
         for artifact in artifacts:
             next_segment = f'\n\nText segment:\n"""\n{artifact.value}\n"""'
```
Comment (vasinov) [inline griptape/engines/query/vector_query_engine.py:39]: We should be able to tuck "Text segment" into a template as well and just use a for loop there:

```
{% for segment in text_segments %}
...
{% endfor %}
```

File: griptape/engines/query/vector_query_engine.py [hunk 2]
```diff
@@ -28,28 +33,26 @@ def query(
...
 
-            if tokenizer.token_count(prefix + next_segment + question + answer) > tokenizer.max_tokens:
+            message = self.template_generator.render(
+                metadata=metadata or "-",
```
Comment (vasinov) [inline griptape/engines/query/vector_query_engine.py:42]: Probably can keep it as `metadata` and not include it in the prompt at all if it's none:

```
{% if metadata %}Metadata: {{ metadata }}{% endif %}
```

**Review by vasinov — CHANGES_REQUESTED** [review]
Thanks for the PR! This is a much needed improvement.

**Review by vasinov — CHANGES_REQUESTED** [review]
This is so great! One last thing: could you please add a couple of unit tests with a custom template just to make sure text segments are appended correctly? Thanks!

**Comment by joelwiesmann** [comment]
> This is so great! One last thing: could you please add a couple of unit tests with a custom template just to make sure text segments are appended correctly? Thanks!

Sure! Will have a look on that tomorrow. Will consider this in the future from the beginning :-).

**Comment by joelwiesmann** [comment]
@vasinov I'm not sure what you meant by "custom template", so I added a test case to see, whether the provided metadata / text segments & question end up in the created prompt at all. This case is expected to work with whatever query template is used.

**Pull request merged**